### ISSUE TITLE
Should data be updated manually in csv files or we have an API we can refer to and get latest data?

### ISSUE THREAD

**Issue by VigneshVaidyanathan** [aalises/age-of-empires-II-api#21]

**Comment by aalises** [comment]
Yes, the data is static! I just extracted it from https://aoestats.io/

**Comment by VigneshVaidyanathan** [comment]
Do you think this can be automated? I tried to find the data from https://aoestats.io/ but it has only match data. I am curious to know if you had any backend service running to parse the raw data into the form you are using from the CSV or is it something we can get from somewhere.

Also, how difficult is it to update the details, I would like to help but I am not sure how to get anywhere.

I want to see this applies_to data mainly in a Tech tree format so it can be widely used for beginners to know what techs will affect which units/buildings.

**Comment by shiibbyy** [comment]
I too am wondering where you got the rest of the data from as on aoestats.io I can only find the player-match data and nothing relating to civs, units or tech. 

Really interested in this API and would love to expand upon it, just struggling to find a starting point

**Comment by simonsan** [comment]
What about this https://github.com/SiegeEngineers/aoe2techtree/blob/master/data/data.json
or https://github.com/SiegeEngineers/halfon/blob/master/data/units_buildings_techs.de.json

**Comment by aalises** [comment]
Hello legends! Just stumbled back into the repository after a while, so I'm not sure of the status of it. Anyways, there's much to improve on it so feel free to submit some PRs or fork it themselves!

Dynamically getting the data from a maintained source would indeed be a nice improvement. I would imagine something along the lines of (for whoever wants to give it a swing)

- Create a script that pulls the latest data from a maintained source (can be an API, some static JSON files, scraping a website...)
- Sanitize and do some data engineering on the source to adapt it to our data model
- Insert it into our database

We could then run that script on demand as a pre-requisite step when building the application and jazz 🎷, up-to-date data for the API!
